### DISCUSSION TITLE
Partial FocusAdorner for a ControlThemed control

### DISCUSSION THREAD

**Discussion by flarive** [AvaloniaUI/Avalonia#13998] [General]
Hello,

I have a custom ControlTheme for a ToggleSwitch control.
This ToggleSwitch control displays the toggle and a text next to the toggle.

![1](https://github.com/AvaloniaUI/Avalonia/assets/7724904/5baeccce-e059-4ce7-9a26-552bc08b00e0)

I would like to add a custom FocusAdorner for that control in my custom theme.

```
  <ControlTheme x:Key="{x:Type ToggleSwitch}" TargetType="ToggleSwitch">
...
... cutted stuffs here
...
    <Style Selector="^:is(Control)">
      <Setter Property="FocusAdorner">
        <FocusAdornerTemplate>
          <Border BorderThickness="2" CornerRadius="100"
                  BorderBrush="Orange"/>
        </FocusAdornerTemplate>
      </Setter>
    </Style>
  </ControlTheme>
```

But the FocusAdorner is taking all control (toggle + text), it's ugly, i would like the FocusAdorner to tale only a part of the control (just the toggle and not the text)

What i have :
![2](https://github.com/AvaloniaUI/Avalonia/assets/7724904/b3ea8436-c9f8-481a-a8a7-5f6a1d3660b3)

What i would like :
![3](https://github.com/AvaloniaUI/Avalonia/assets/7724904/cd7d3b17-8f99-4f93-968f-b723d4a6cede)

How could i do that ?

@maxkatz6 any idea :) ?

Thank you sooo much

Eviral

**Comment by timunie** [comment]
@flarive I think you can get AdornedElement from the Adorner and from there you may be able to get the toggle switch part. However I feel like setting the SwitchWidth via DynamicResource which can be re-used may be a simpler approach if that would work for you.

**Reply by flarive** [reply]
Thanks a lot Tim
Maxkatz6 solution works

**Comment by maxkatz6** [comment]
1. Disable FocusAdorner https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Themes.Fluent/Controls/TextBox.xaml#L108
2. Handle custom `:focus` pseudoclass in your way: https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Themes.Fluent/Controls/TextBox.xaml#L218

**Reply by flarive** [reply]
Works perfectly...
I like Avalonia so much because there is never any real blocker, everytime there is a solution to be able to acheive exactly what we want.
Thanks again !
